dashboard/test/controllers/pd/workshop_user_management_controller_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'test_helper'
+
+class Pd::WorkshopUserManagementControllerTest < ActionController::TestCase
+  self.use_transactional_test_case = true
+  setup_all do
+    @workshop_admin = create :workshop_admin
+    @teacher = create :teacher
+    @student = create :student
+    @facilitator = create :facilitator
+    @facilitator_with_course = create(:pd_course_facilitator, course: Pd::Workshop::COURSE_CSF).facilitator
+  end
+
+  def self.test_workshop_admin_only(method, action, success_response, params = nil)
+    test_user_gets_response_for action, user: :student, method: method, params: params, response: :forbidden
+    test_user_gets_response_for action, user: -> {@teacher}, method: method, params: params, response: :forbidden
+    test_user_gets_response_for action, user: -> {@workshop_admin}, method: method, params: params, response: success_response
+  end
+
+  test_redirect_to_sign_in_for :facilitator_courses_form
+  test_workshop_admin_only :get, :facilitator_courses_form, :success
+
+  test 'find facilitator for non-existent email displays no facilitator error' do
+    sign_in @workshop_admin
+    post :facilitator_courses_form, params: {search_term: 'nonexistent@example.net'}
+    assert_select '.alert-success', 'Facilitator not found'
+  end
+
+  test 'find facilitator for non-existent id displays no facilitator error' do
+    sign_in @workshop_admin
+    post :facilitator_courses_form, params: {search_term: -999}
+    assert_select '.alert-success', 'Facilitator not found'
+  end
+
+  test 'find facilitator by id for existing facilitator displays facilitator email' do
+    sign_in @workshop_admin
+    post :facilitator_courses_form, params: {search_term: @facilitator.id}
+    assert_select 'td', @facilitator.email
+  end
+
+  test 'find facilitator by email for existing facilitator displays facilitator id' do
+    sign_in @workshop_admin
+    post :facilitator_courses_form, params: {search_term: @facilitator.email}
+    assert_select 'td', text: @facilitator.id.to_s
+  end
+
+  test 'assign course to facilitator assigns course' do
+    sign_in @workshop_admin
+    get :assign_course, params: {facilitator_id: @facilitator.id, course: Pd::Workshop::COURSE_ECS}
+    assert_redirected_to action: :facilitator_courses_form, params: {search_term: @facilitator.id}
+    assert @facilitator.courses_as_facilitator.exists?(course: Pd::Workshop::COURSE_ECS), "#{Pd::Workshop::COURSE_ECS} was not assigned to Facilitator - #{@facilitator.email}"
```

Comment:
Didn't we have an `assert_creates Pd::CourseFacilitator` here? Anyway we should. 

----
Alternatively, you can mock the underlying `User.course_as_facilitator=` and make sure it's called with the right params. However the `@facilitator` instance in the controller returned by `User.find` on line 26 will be a different instance than the `@facilitator` here in the test, so you'd also have to stub that:

```ruby
sign_in @workshop_admin
mock_facilitator = mock
User.expects(:new).with(@facilitator.id).returns(mock_facilitator)
mock_facilitator.expects(:course_as_facilitator=).with(Pd::Workshop::COURSE_ECS)

get :assign_course, params: {facilitator_id: @facilitator.id, course: Pd::Workshop::COURSE_ECS}
assert_redirected_to action: :facilitator_courses_form, params: {search_term: @facilitator.id}
```

Note: the mock approach won't actually create the db entry so in that case you won't need the assert on line 50.
Yes this is less concise than what you have, but it's arguably a more pure unit test because it's only testing the intended controller action without side effects. It assumes `User.course_as_facilitator=` works, of which we can be confident since that has its own unit tests :)

This test will succeed only if the expected method calls with params, and no unexpected methods on `mock_facilitator`, are made in addition to the explicit assert.

See the [mocha](http://gofreerange.com/mocha/docs/) docs for more info.

Your choice. Since the dependency is so small I'm ok relying on it.